libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/WayNameView.java

```diff
@@ -41,9 +55,10 @@ public String retrieveWayNameText() {
   }
 
   public void updateVisibility(boolean isVisible) {
-    int visibility = isVisible ? VISIBLE : INVISIBLE;
-    if (getVisibility() != visibility) {
-      setVisibility(visibility);
+    if (isVisible && !TextUtils.isEmpty(wayNameText.getText())) {
```

Comment:
Could we extract this piece of code into a `private` method and reuse it in `updateWayNameText` and remove some duplication? This also applies to legacy `WayNameView`.

Reply:
@RingerJK / @Guardiola31337 : after re-think of it, update the visibility inside of `updateWayNameText` call would be problematic and hard to maintain a clear visibility flow. I removed the change within `updateWayNameText`